diff --git a/index.php b/index.php
index e099737..8dae1f4 100644
--- a/index.php
+++ b/index.php
@@ -14,7 +14,7 @@
  */
 
 
-$path = __DIR__;      // The base directory for the photos, defaults to the current directory
+$path = getcwd();      // The base directory for the photos, defaults to the current directory
 $thumbnailSize = 300; // size in pixels
 $bgcolor = '#d0d5ee'; // background color
 
